### PR TITLE
Doc updates; tablet throttler: check-self

### PR DESCRIPTION
**Pull request by shlomi-noach**
Documenting the changes in https://github.com/vitessio/vitess/pull/7319: a new API endpoint `/throttler/check-self` and what it means.

### CODE REVIEW

**Comment by shlomi-noach** [comment]
Relevant preview link: https://deploy-preview-686--vitess.netlify.app/docs/reference/features/tablet-throttler/

**Comment by deepthi** [comment]
Let us hold this until 9.0 is out?

**Comment by shlomi-noach** [comment]
> Let us hold this until 9.0 is out?

Sure!

**Comment by shlomi-noach** [comment]
This should be merged now, as the code has been merged.

File: content/en/docs/reference/features/tablet-throttler.md [hunk 1]
```diff
@@ -60,6 +67,12 @@ The user may override the heartbeat interval by sending `-heartbeat_interval` fl
 
 Thus, the aggregated interval can be off, by default, by some `500ms`. This makes it inaccurate for evaluations that require high resolution lag evaluation. This resolution is sufficient for throttling purposes.
 
+### Self health
+
+Each tablet to its own, runs a local health check against its backend database, again in the form of evaluating replication lag from `_vt.heartbeat`. Intervals are identical to the cluster health interval illustrated above.
```
Comment (deepthi) [inline content/en/docs/reference/features/tablet-throttler.md]:
```suggestion
Each tablet runs a local health check against its backend database, again in the form of evaluating replication lag from `_vt.heartbeat`. Intervals are identical to the cluster health interval illustrated above.
```

File: content/en/docs/reference/features/tablet-throttler.md [hunk 2]
```diff
@@ -78,27 +91,27 @@ When the throttler sees no relevant replicas in the shard, it allows writes by r
 
 ## Configuration
 
-
-- The throttler is currently disabled by default. Use the `vttablet` option `-enable-lag-throttler` to enable the throttler.
-  When the throttler is disabled, it still serves `/throttler/check` API and responds with `HTTP 200 OK` to all requests.
+- The throttler is currently **disabled** by default. Use the `vttablet` option `-enable-lag-throttler` to enable the throttler.
```
Comment (deepthi) [inline content/en/docs/reference/features/tablet-throttler.md:94]: Can we keep the flags consistent and always use underscores? we have `enable-lag-throttler` but `throttle_threshold`.

**Review by deepthi — APPROVED** [review]
A couple of suggestions. You can go ahead and merge after addressing them.

File: content/en/docs/reference/features/tablet-throttler.md (hunk 2 above)
Comment (shlomi-noach) [reply content/en/docs/reference/features/tablet-throttler.md:94]: Right! So this is all over the place, we have dashes and underscored spread across many flags. I'm just wondering how do we go about this with existing flags? Let me see if there's something in golang that treats both as the same, as it it's an uppercase/lowercase thing.

Comment (shlomi-noach) [reply content/en/docs/reference/features/tablet-throttler.md:94]: Noting that this is not a documentation issue -- the flag is named `-enable-lag-throttler` and we should normalize the flag name in code.

Comment (shlomi-noach) [reply content/en/docs/reference/features/tablet-throttler.md:94]: A bit of digging brought up [this](https://github.com/golang/go/issues/35761#issuecomment-559207341). So what I'll do is for some variables support both dash-based and underscore-based names.

**Pull request merged**